com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs

```diff
@@ -241,5 +244,35 @@ internal void UpdateAgentPolicy()
             }
             agent.ReloadPolicy();
         }
+
+        internal void UpdateAnalytics(IList<ISensor> sensors, ActionSpec actionSpec)
+        {
+            switch (m_BehaviorType)
```

Comment:
I don't love this, but it follows the same structure as GeneratePolicy(). We can't send the event in GeneratePolicy() because we don't have the sensors available.

I could also move the call to InferenceModelSet to the first time that BarracudaPolicy.RequestDecision is called.

Reply:
Yeah, I don't like this either. What if we add the sensors to GeneratePolicy or what if we check the type of m_Brain right after the call to `m_Brain = m_PolicyFactory.GeneratePolicy(combinedActionSpec, Heuristic);`. If it is a BarracudaPolicy then send the event. I think it is better than trying to copy the internal behavior of GeneratePolicy.

Reply:
Discussed offline, adding to BarracudaPolicy.RequestDecision